src/micro_http/src/connection.rs

```diff
@@ -171,20 +177,24 @@ impl<T: Read + Write> HttpConnection<T> {
         }
     }
 
-    // Parses bytes in `buffer` for header fields.
-    // Returns `false` if there are no more bytes to be parsed in the buffer.
+    /// Parses bytes in `buffer` for header fields.
+    /// Returns `false` if there are no more bytes to be parsed in the buffer.
+    ///
+    /// # Errors
+    /// `ParseError` is returned if unable to parse header or line longer than BUFFER_SIZE.
     fn parse_headers(
         &mut self,
         line_start_index: &mut usize,
         end_cursor: usize,
     ) -> Result<bool, ConnectionError> {
         match find(&self.buffer[*line_start_index..end_cursor], &[CR, LF]) {
-            // We have found the end of the headers.
             // `line_start_index` points to the end of the most recently found CR LF
             // sequence. That means that if we found the next CR LF sequence at this index,
             // they are, in fact, a CR LF CR LF sequence, which marks the end of the header
             // fields, per HTTP specification.
             Some(0) => {
+                // Our relative line end index is zero. this means we have reached end of the
```

Comment:
relative line end index is confusing, what was 
```suggestion
                // Our `relative_line_end_index is zero; this means we have reached end of the
```